script.js

```diff
@@ -1,4 +1,111 @@
+var compwin = 0;
+var userwin = 0;
+var draw = 0;
+
+var currentGameMode = "waiting for user name";
+var userName = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  // do this if game mode is waiting for username
+  if (currentGameMode == "waiting for user name") {
+    //set the name
+    userName = input;
+
+    // Once username is inputted, switch mode
+    currentGameMode = "SPS mode";
+
+    myOutputValue =
+      "Hello " +
+      userName +
+      "! You can start playing the game by entering one of the 3 objects: scissors, paper or stone";
+  } else if (currentGameMode == "SPS mode") {
+    var randomnumber = number();
+    var SorPorS = numbertoSPS(randomnumber);
+
+    console.log(SorPorS);
+
+    compwin = compwin + 1;
+    myOutputValue =
+      "The computer chose " +
+      SorPorS +
+      ". <br > You chose " +
+      input +
+      ". <br >You lose! Bummer " +
+      userName +
+      "! The computer won " +
+      compwin +
+      " times against you!";
+
+    // define the winning conditions and draw conditions. If it isn't then the user loses.
+    if (input == SorPorS) {
+      draw = draw + 1;
+      myOutputValue =
+        "The computer chose " +
+        SorPorS +
+        ". <br > You chose " +
+        input +
+        ". <br >It is a draw! Try again " +
+        userName +
+        "! You drawed " +
+        draw +
+        " times with the computer";
+    }
```

Comment:
Good realisation that we don't have to enumerate all the conditions for the draw